File: gptcomet/llms/kimi.py
from typing import Any

from .openai import OpenaiLLM


class KimiLLM(OpenaiLLM):
    def __init__(self, config: dict[str, Any]):
        super().__init__(config)

        self.api_base = config.get("api_base") or "https://api.kimi.ai/v1"
        self.model = config.get("model") or "moonshot-v1-8k"
        self.completion_path = self.completion_path or "chat/completions"
        self.answer_path = self.answer_path or "choices.0.message.content"

    @classmethod
    def get_required_config(cls) -> dict[str, tuple[str, str]]:
        return {
            "api_base": ("https://api.kimi.ai/v1", "Enter Kimi API Base URL"),
            "model": ("moonshot-v1-8k", "Enter model name"),
            "api_key": ("", "Enter API key"),
            "max_tokens": ("1024", "Enter max tokens"),
        }

File: internal/llm/kimi.go
package llm

import (
	"context"
	"net/http"

	"github.com/belingud/gptcomet/pkg/config"
	"github.com/belingud/gptcomet/pkg/types"
)

const (
	DefaultKimiAPIBase = "https://api.moonshot.cn/v1"
	DefaultKimiModel   = "moonshot-v1-8k"
)

// KimiLLM implements the LLM interface for Kimi
type KimiLLM struct {
	*OpenAILLM
}

// NewKimiLLM creates a new KimiLLM
func NewKimiLLM(config *types.ClientConfig) *KimiLLM {
	BuildStandardConfigSimple(config, DefaultKimiAPIBase, DefaultKimiModel)

	return &KimiLLM{
		OpenAILLM: NewOpenAILLM(config),
	}
}

func (k *KimiLLM) Name() string {
	return "kimi"
}

// GetRequiredConfig returns provider-specific configuration requirements
func (k *KimiLLM) GetRequiredConfig() map[string]config.ConfigRequirement {
	return map[string]config.ConfigRequirement{
		"api_base": {
			DefaultValue:  DefaultKimiAPIBase,
			PromptMessage: "Enter Kimi API base",
		},
		"api_key": {
			DefaultValue:  "",
			PromptMessage: "Enter API key",
		},
		"model": {
			DefaultValue:  DefaultKimiModel,
			PromptMessage: "Enter model name",
		},
		"max_tokens": {
			DefaultValue:  "1024",
			PromptMessage: "Enter max tokens",
		},
	}
}

// MakeRequest makes a request to the Kimi API
func (k *KimiLLM) MakeRequest(ctx context.Context, client *http.Client, message string, stream bool) (string, error) {
	return k.BaseLLM.MakeRequest(ctx, client, k, message, stream)
}
